### PR TITLE
Enforce min master nodes in test cluster

### PR DESCRIPTION
**Pull request by bleskes**
In order to start clusters with min master nodes set without setting `discovery.initial_state_timeout`, #21846 has changed the way we start nodes. Instead to the previous serial start up, we now always start the nodes in an async fashion (internally). This means that starting a cluster is unsafe without `min_master_nodes` being set. We should therefore make it mandatory.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java [hunk 1]
```diff
@@ -589,12 +590,14 @@ private NodeAndClient buildNode(int nodeId, long seed, Settings settings,
...
         if (autoManageMinMasterNodes) {
-            assert finalSettings.get(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey()) == null :
+            assert finalSettings.get(DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey()) == null :
                 "min master nodes may not be set when auto managed";
             finalSettings
                 // don't wait too long not to slow down tests
                 .put(ZenDiscovery.MASTER_ELECTION_WAIT_FOR_JOINS_TIMEOUT_SETTING.getKey(), "5s")
-                .put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), defaultMinMasterNodes);
+                .put(DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), defaultMinMasterNodes);
+        } else if (finalSettings.get(DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey()) == null){
```
Comment (ywelsch) [inline test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java]: space missing between ) and {

File: test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java [hunk 2]
```diff
@@ -589,12 +590,14 @@ private NodeAndClient buildNode(int nodeId, long seed, Settings settings,
...
+            throw new IllegalStateException(DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey() + " must be configured");
```
Comment (ywelsch) [inline test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java]: nit: IllegalArgumentException ;-) You're checking the `settings` argument here, not the state of InternalTestCluster.

**Review by ywelsch — APPROVED** [review]
Looks good! (Left 2 nits)

**Comment by bleskes** [comment]
thx @ywelsch . This is now backported to 5.x & 5.1 as well

**Pull request merged**